### PR TITLE
Feature/apns

### PR DESCRIPTION
**Pull request by wenjdu**
Adding Apns to SDK.

### CODE REVIEW

File: AlpsSDK/NotificationManager.swift [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                UIApplication.shared.registerForRemoteNotifications()
+            }
+        }
+            // iOS 9 support
+        else if #available(iOS 9, *) {
+            UIApplication.shared.registerUserNotificationSettings(UIUserNotificationSettings(types: [.badge, .sound, .alert], categories: nil))
+            self.getNotificationSettings()
+            UIApplication.shared.registerForRemoteNotifications()
+        }
+            // iOS 8 support
```
Comment (maciejburda) [inline AlpsSDK/NotificationManager.swift:49]: I think at some point, we should stop supporting.

Also note that 8 and 9 have exactly the same code.

iOS9+ in most cases is over 99% now.
https://david-smith.org/iosversionstats/

File: AlpsSDK/ContextManager.swift [hunk 2]
```diff
@@ -458,59 +497,53 @@ class ContextManager: NSObject, CLLocationManagerDelegate {
                    
                     if gap > 5 * 60 * 1000 {
                         t.removeValue(forKey: id)
-                        for i in 0...3 {
-                            switch i{
-                            case 0:
+                        for proximity in CLProximity.allValues {
```
Comment (maciejburda) [inline AlpsSDK/ContextManager.swift:500]: it seems like a lot of fors and switches.

Maybe we can come up with some nice functional code that would make it much shorter (better to maintain)

File: AlpsSDK/NotificationManager.swift [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+//  Copyright © 2017 Alps. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import UserNotifications
+
+class NotificationManager : NSObject, UNUserNotificationCenterDelegate {
+    var alpsManager : AlpsManager
+    var deviceToken : Data? = nil
```
Comment (maciejburda) [inline AlpsSDK/NotificationManager.swift]: no need to have `= nil` at the end.

File: AlpsSDK/NotificationManager.swift [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    var deviceTokenString : String = ""
```
Comment (maciejburda) [inline AlpsSDK/NotificationManager.swift]: `var deviceTokenString = ""` is shorter to read. swift can figure out the type itself.

File: AlpsSDK/NotificationManager.swift [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    var seenError : Bool = false
```
Comment (maciejburda) [inline AlpsSDK/NotificationManager.swift]: same as above.

File: AlpsSDK/NotificationManager.swift [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+        } else {
+            // Fallback on earlier versions
+            print("************PREVIOUS VERSION")
+        }
+    }
+    
+    // Called when AppDelegate func didRegisterForRemoteNotificationsWithDeviceToken
+    //
+    func registerDeviceToken(deviceToken: Data, deviceTokenString : String){
+        print("REGISTERED DEVICE TOKEN")
```
Comment (maciejburda) [inline AlpsSDK/NotificationManager.swift]: we should avoid printing a lot into console without control over it.

Some kind of logger seems like a nice task.

File: AlpsSDK/ContextManager.swift (hunk 2 above)
Comment (wenjdu) [reply AlpsSDK/ContextManager.swift:500]: 👍  I agree !

File: AlpsSDK/NotificationManager.swift (hunk 1 above)
Comment (wenjdu) [reply AlpsSDK/NotificationManager.swift:49]: Ok ! Yes I also agree with stop supporting some ios x previous version. But wanted to add the most and correct later 👍

File: AlpsSDK/NotificationManager.swift (hunk 6 above)
Comment (wenjdu) [reply AlpsSDK/NotificationManager.swift]: Ok using NSLog instead.

File: AlpsSDK/NotificationManager.swift (hunk 1 above)
Comment (maciejburda) [reply AlpsSDK/NotificationManager.swift:49]: @tharpa what version we need to support?

**Comment by maciejburda** [comment]
We dediced to add this code after refactoring.

Closing PR as unmerged.

**Pull request closed without merging**